lib/spack/spack/util/compression.py

```diff
@@ -186,47 +182,47 @@ def _lzma_decomp(archive_file):
         with open(archive_out, "wb") as ar:
             with lzma.open(archive_file) as lar:
                 shutil.copyfileobj(lar, ar)
+        return archive_out
     else:
-        if sys.platform == "win32":
-            return _7zip(archive_file)
-        else:
-            return _xz(archive_file)
+        return _xz(archive_file)
 
 
-def _win_compressed_tarball_handler(archive_file):
+def _win_compressed_tarball_handler(helper):
```

Comment:
I think `decompressor` would be a clearer name than `helper`, since that's what it is doing.

Reply:
Good point! Will amend.